src/components/Visualization.tsx

```diff
@@ -51,10 +53,54 @@ const Visualization = () => {
     shallow,
   );
 
-  const flows = useFlowsStore((state) => state.flows);
+  const { settings, setSettings } = useSettingsStore((state) => state);
+  const { setSourceCode, setSyncedSourceCode } = useIntegrationSourceStore((state) => state);
```

Comment:
oh, is not a stupid question by any means :smiley:, actually, you're right, it shouldn't be necessary at all. Let me check what the `shallow` function does in case of a `function`.